### ISSUE TITLE
Bug: error with retrieving input dim in guided_grad_cam.py? 

### ISSUE THREAD

**Issue by ybzhaoplus** [pytorch/captum#667]
https://github.com/pytorch/captum/blob/43acfcd320c44e8e469bf5276af9d55cf62145aa/captum/attr/_core/guided_grad_cam.py#L211

If I understand corretly, the parameter here should be: 
inputs[i].shape[1:], rather then inputs[i].shape[2:]

For inputs, the shape is N-C-H-W, 
For inputs[i], the shape is C-H-W, thus its dim (H-W) could be inputs[i].shape[1:]

As when I run with Guided GradCAM, it keeps warning: "Couldn't appropriately interpolate GradCAM attributions for some input tensors, returning empty tensor for corresponding attributions."

**Comment by bilalsal** [comment]
Hi @ybzhaoplus,

thank you for commenting on the implementation!

The current implementation should correctly index the H and W dimensions, because an input tensor is first converted into a tuple:
https://github.com/pytorch/captum/blob/43acfcd320c44e8e469bf5276af9d55cf62145aa/captum/_utils/common.py#L153

Is it possible that some of the tensors you are passing have a dimensionality less than 4? Or maybe 
Hope this helps

**Comment by ybzhaoplus** [comment]
Thanks for getting back.

I checked the inputs, and the dimensionality is 4.

I bypass this step by implementing guided_grad_cam by myself with guided_backprop and gradcam in captum. I use inputs.shape[2:], and it works.

**Comment by bilalsal** [comment]
Thanks for the clarification, @ybzhaoplus.

We are interested in understanding this issue further. Do you mind sharing a Colab notebook illustrating the exception and the warning you got?

Thanks!

**Comment by NarineK** [comment]
@ybzhaoplus is this still an open issue ?

**Comment by Tianmaru** [comment]
I can't get Guided GradCAM to work in Captum 0.6.0 either, although GradCAM itself works perfectly fine. I tried different input shapes (with and without batch dimension) and interpolation modes, but it always results in the following error:
`UserWarning: Couldn't appropriately interpolate GradCAM attributions for some input tensors, returning empty tensor for corresponding attributions.`

**Comment by craymichael** [comment]
Hi, I was unable to reproduce the issue. Since this issue is old, I will keep it closed for now. But if you're still having it, please reopen and sorry for the delay.

**Comment by craymichael** [comment]
Checked again with Captum v0.9.0. Minimal check:

```python
import torch
import torch.nn as nn
from captum.attr import GuidedGradCam

class Net(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = nn.Conv3d(1, 2, 3, padding=1)
        self.fc = nn.Linear(2, 2)
    def forward(self, x):
        y = torch.relu(self.conv(x))
        return self.fc(y.mean(dim=(2, 3, 4)))

model = Net().eval()
x = torch.randn(1, 1, 4, 5, 6)
attr = GuidedGradCam(model, model.conv).attribute(x, target=0)
assert attr.shape == x.shape
```

Behavior: attribution returns full input-shaped output with no errors or interpolation warning locally.
